src/sources/syslog_remap.rs

```diff
@@ -0,0 +1,97 @@
+use crate::sources::socket::SocketConfig;
+use crate::{
+    config::{
+        log_schema, DataType, GenerateConfig, Resource, SourceConfig, SourceContext,
+        SourceDescription,
+    },
+    transforms::remap::{Remap, RemapConfig},
+    Pipeline,
+};
+use bytes::Bytes;
+use indoc::indoc;
+use futures::StreamExt;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct SyslogRemapConfig {
+    // Config settings we may need access to
+    // host_key: Option<String>,
+    #[serde(flatten)]
+    original_config: SocketConfig,
+}
+
+inventory::submit! {
+    SourceDescription::new::<SyslogRemapConfig>("syslog_remap")
+}
+
+impl GenerateConfig for SyslogRemapConfig {
+    fn generate_config() -> toml::Value {
+        SocketConfig::generate_config()
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "syslog_remap")]
+impl SourceConfig for SyslogRemapConfig {
+    async fn build(&self, mut cx: SourceContext) -> crate::Result<super::Source> {
+        let conf = RemapConfig {
+            source: indoc! {r#"
+                structured = parse_syslog!(.message)
+                . = merge(., structured)
+            "#}
+            .to_string(),
+            drop_on_abort: false,
+            drop_on_error: false,
+        };
+        let tf = Remap::new(conf).unwrap();
+        let (to_transform, rx) = Pipeline::new_with_buffer(100, vec![Box::new(tf)]);
+        let out = cx.out;
+        cx.out = to_transform;
+        tokio::spawn(async move {
+            rx.map(|mut event| {
+                event
+                    .as_mut_log()
+                    .insert(log_schema().source_type_key(), Bytes::from("syslog_remap"));
+                Ok(event)
+            })
```

Comment:
This is an interesting bit I hadn't thought of. Maybe we should make this kind of tagging something that happens from the source context and/or pipeline as well? That way we could again just modify the context and pass it down to keep this simpler.

Reply:
As we do it for every source it would makes sense to do it on a general basis. From what I can tell if we want to leverage the pipeline logic we could use something like the `AddFields` transform in the `inlines` vec.